blocks/results-list-block/features/results-list/default.jsx

```diff
@@ -128,39 +128,54 @@ class ResultsList extends Component {
             transform: (data) => fetchStoriesTransform(data, storedList),
           },
         });
-        // Hide button if no more stories to load
-        if ((storedList.content_elements.length + size) >= count) {
-          this.state.seeMore = false;
-        }
-      }
-    } else {
-      if (listContentConfig.contentService === 'content-api-collections') {
-        // Count all items from collection based on contentConfigValues conditions
+
         const query = { ...contentConfigValues };
-        // By default collection limit 20 items
-        query.size = '20';
+        from = parseInt(contentConfigValues.from, 10) || 0;
+        query.from = String(from + size + 1);
         this.fetchContent({
-          collection: {
+          seeMore: {
             source: contentService,
             query,
+            transform: (data) => {
+              if (!data || data.content_elements.length === 0) {
```

Comment:
We should probably check that data has `content_elements` before  we attempt to get the length